src/structures/MessageEmbed.js

```diff
@@ -345,15 +345,38 @@ class MessageEmbed {
     return this;
   }
 
+  /**
+   * The options to provide for setting an author for a {@link MessageEmbed}.
+   * @typedef {Object} EmbedAuthorData
+   * @property {string} name The name of this author.
+   * @property {string} [url] The URL of this author.
+   * @property {string} [iconURL] The icon URL of this author.
+   */
+
+  // TODO: Remove the deprecated code in the following method and typings.
   /**
    * Sets the author of this embed.
-   * @param {string} name The name of the author
-   * @param {string} [iconURL] The icon URL of the author
-   * @param {string} [url] The URL of the author
+   * @param {string|EmbedAuthorData|null} options The options to provide for the author.
+   * A string may simply be provided if only the author name is desirable.
+   * Provide `null` to remove the author data.
+   * @param {string} [deprecatedIconURL] The icon URL of this author.
+   * <warn>This parameter is **deprecated**. Use the `options` parameter instead.</warn>
+   * @param {string} [deprecatedURL] The URL of this author.
+   * <warn>This parameter is **deprecated**. Use the `options` parameter instead.</warn>
    * @returns {MessageEmbed}
    */
-  setAuthor(name, iconURL, url) {
-    this.author = { name: Util.verifyString(name, RangeError, 'EMBED_AUTHOR_NAME'), iconURL, url };
+  setAuthor(options, deprecatedIconURL, deprecatedURL) {
+    if (options === null) {
+      this.author = {};
+      return this;
+    }
```

Comment:
Those lines should be moved after the typeof check to keep the behaviour of the overload.

Otherwise `.setAuthor(null, url, iconURL)` will make the embed not throw.